source/common/network/lc_trie.h

```diff
@@ -21,6 +21,13 @@ namespace Envoy {
 namespace Network {
 namespace LcTrie {
 
+/**
+ * Maximum number of nodes an LC trie can hold.
+ * @note If the size of LcTrieInternal::LcNode::address_ ever changes, this constant
+ *       should be changed to match.
+ */
+static const size_t MaxLcTrieNodes = (1 << 20);
```

Comment:
Could also make this `constexpr` instead of `static const` to indicate its intended inlined use.